lib/date_range.rb

```diff
@@ -0,0 +1,32 @@
+require "date"
+
+module Hotel
+  class DateRange
+    attr_accessor :check_in, :check_out
+    attr_reader :nights
+
+    def initialize(check_in:, check_out:)
+      if check_out != nil
+        if check_out > check_in
+          @check_in = check_in
+          @check_out = check_out
+        else
+          raise ArgumentError, "Start date must be before end date"
+        end
+      end
+      @nights = (@check_out - @check_in).to_i
+    end
+
+    def overlap?(new_date_range)
+      if (new_date_range.check_in >= self.check_out || new_date_range.check_out <= self.check_in)
+        return false
+      else
+        return true
+      end
```

Comment:
This is very minor but this block of code could be simplified to what I have below and it would do the same thing.
```suggestion
      return !(new_date_range.check_in >= self.check_out || new_date_range.check_out <= self.check_in)
```

Reply:
After looking at it though, I think the way you have it may be easier to read and understand.